engine/schema/src/main/java/com/cloud/storage/dao/VolumeDaoImpl.java

```diff
@@ -841,4 +866,13 @@ public List<VolumeVO> listByIds(List<Long> ids) {
         sc.setParameters("idIN", ids.toArray());
         return listBy(sc, null);
     }
+
+    @Override
+    public List<VolumeVO> listByPoolIdVMStatesNotInCluster(long clusterId, List<VirtualMachine.State> states, long poolId) {
+        SearchCriteria<VolumeVO> sc = volumePoolNotInClusterSearch.create();
+        sc.setParameters("poolId", poolId);
+        sc.setParameters("states", states);
```

Comment:
Not sure. I think `states` need to be an array here.

Reply:
There was a bug here in the way I was using the conditionName. 
Have fixed and tested. It seems to be working. Please look